WooCommerce/src/main/kotlin/com/woocommerce/android/ui/searchfilter/SearchFilterFragment.kt

```diff
@@ -0,0 +1,152 @@
+            searchFilterItems = navArgs.items,
+            searchHint = navArgs.hint,
+            requestKey = navArgs.requestKey
+        )
+        setupViewStateObserver()
+        setupEventObserver()
+        setupSearchInput()
+        setupSearchList()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun setupViewStateObserver() {
+        viewModel.viewStateLiveData.observe(
+            viewLifecycleOwner,
+            { viewState ->
+                when (viewState) {
+                    is Loaded -> {
+                        showSearchList()
+                        updateSearchList(viewState.searchFilterItems)
+                        setupTitle(viewState.title)
+                        binding.searchEditText.hint = viewState.searchHint
+                    }
+                    is Search -> {
+                        showSearchList()
+                        updateSearchList(viewState.searchFilterItems)
+                        hideEmptyView()
+                    }
+                    is Empty -> {
+                        hideSearchList()
+                        showEmptyView(viewState.searchQuery)
+                    }
+                }.exhaustive
+            }
+        )
+    }
+
+    private fun setupEventObserver() {
+        viewModel.eventLiveData.observe(
+            viewLifecycleOwner,
+            { event ->
+                when (event) {
+                    is ItemSelected -> {
+                        navigateBackWithResult(event.requestKey, event.selectedItemValue)
+                    }
+                }.exhaustive
+            }
+        )
+    }
+
+    private fun setupSearchInput() {
+        binding.searchEditText.apply {
+            doAfterTextChanged {
+                viewModel.onSearch(it.toString())
+            }
+        }
+    }
+
+    private fun setupSearchList() {
+        binding.searchItemsList.apply {
+            layoutManager = LinearLayoutManager(context)
+            searchFilterAdapter = SearchFilterAdapter(
+                onItemSelectedListener = { selectedItem ->
+                    viewModel.onItemSelected(selectedItem)
+                }
+            )
+            adapter = searchFilterAdapter
+            addItemDecoration(
+                AlignedDividerDecoration(
+                    ctx = context,
+                    orientation = DividerItemDecoration.VERTICAL,
+                    alignStartToStartOf = R.id.filterItemName
+                )
+            )
+        }
+    }
+
+    private fun updateSearchList(searchFilterItems: List<SearchFilterItem>) {
+        searchFilterAdapter.setItems(searchFilterItems.toList())
+    }
+
+    private fun showSearchList() {
+        binding.searchItemsList.show()
+    }
+
+    private fun hideSearchList() {
+        binding.searchItemsList.hide()
+    }
+
+    private fun setupTitle(title: String) {
+        activity?.title = title
```

Comment:
Just an FYI that you can override `getFragmentTitle()` in `BaseFragment` to supply the activity title, like this:

```
override fun getFragmentTitle() = navArgs.title
```